src/psyclone/tests/core/access_info_test.py

```diff
@@ -635,3 +640,50 @@ def test_symbol_array_detection(fortran_reader):
     b_is_array = sym_b.is_array_access(access_info=access_info_b,
                                        index_variable="j")
     assert not b_is_array
+
+
+# -----------------------------------------------------------------------------
+def test_variables_access_info_options():
+    '''Test handling of options for VariablesAccessInfo.
+    '''
+    vai = VariablesAccessInfo(options={'COLLECT-ARRAY-SHAPE-READS': True})
+
+    assert vai.options("COLLECT-ARRAY-SHAPE-READS") is True
+    assert vai.options() == {'COLLECT-ARRAY-SHAPE-READS': True}
+    with pytest.raises(InternalError) as err:
+        vai.options("invalid")
+    assert ("Option key 'invalid' is invalid, it must be one of "
+            "['COLLECT-ARRAY-SHAPE-READS']." in str(err.value))
+
+
+# -----------------------------------------------------------------------------
+@pytest.mark.parametrize("function", ["size", "lbound", "ubound"])
+def test_variables_access_info_shape_bounds(fortran_reader, function):
+    '''Test that access to an array using shape, or lbound/ubound can be
+    disables using options
+    '''
+    code = f'''module test
+        contains
+        subroutine tmp()
+          real, dimension(:,:), allocatable:: a
+          integer :: n
+          n = {function}(a, 1)
+        end subroutine tmp
+        end module test'''
+    psyir = fortran_reader.psyir_from_source(code)
+    schedule = psyir.children[0].children[0]
```

Comment:
Hmm - indeed. While I don't like the idea of walk (we really need to put a limit, i.e. abort after you have found X nodes), indeed it is better than my `children[0].children[0]`. I've replaced that in several places now

Reply:
If you only want the first hit (as in this case) you can do `walk(Operation, stop_type=Operation)`

Reply:
Ah, goo suggestion. I still decided to just leave it the way it is, because it is shorter and there is no performance issue since there is only one statement or so :)

Reply:
OK.